### PR TITLE
Added Test-AzureRmNetworkWatcherConnectivity cmdlet

### PR DESCRIPTION
**Pull request by irrogozh**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network/NetworkWatcher/TestAzureNetworkWatcherConnectivityCommend.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+             Mandatory = true,
+             ValueFromPipeline = true,
+             HelpMessage = "The network watcher resource.",
+             ParameterSetName = "SetByResource")]
+        public PSNetworkWatcher NetworkWatcher { get; set; }
+
+        [Alias("Name")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/NetworkWatcher/TestAzureNetworkWatcherConnectivityCommend.cs]: @irrogozh this needs to be `ValueFromPipelineByPropertyName`

File: src/ResourceManager/Network/Commands.Network/help/Test-AzureRmNetworkWatcherConnectivity.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
+---
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/help/Test-AzureRmNetworkWatcherConnectivity.md]: @irrogozh you will also need to add a reference to this file in [`AzureRM.Network.md`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md)

(just follow the format of all other file references)

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/NetworkWatcherAPITests.cs [hunk 3]
```diff
@@ -74,5 +74,12 @@ public void TestFlowLog()
         {
             NetworkResourcesController.NewInstance.RunPsTest("Test-FlowLog");
         }
+
+        [Fact(Skip = "Rerecord tests")]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/NetworkWatcherAPITests.cs]: @irrogozh why is this test being skipped?

File: src/ResourceManager/Network/Commands.Network/NetworkWatcher/TestAzureNetworkWatcherConnectivityCommend.cs (hunk 1 above)
Comment (irrogozh) [inline src/ResourceManager/Network/Commands.Network/NetworkWatcher/TestAzureNetworkWatcherConnectivityCommend.cs]: Fixed

File: src/ResourceManager/Network/Commands.Network/help/Test-AzureRmNetworkWatcherConnectivity.md (hunk 2 above)
Comment (irrogozh) [inline src/ResourceManager/Network/Commands.Network/help/Test-AzureRmNetworkWatcherConnectivity.md]: Done. Submit PR with changes.

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/NetworkWatcherAPITests.cs (hunk 3 above)
Comment (irrogozh) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/NetworkWatcherAPITests.cs]: 1. Have a problem with starting recording session (not specific to my test)
2. Can't record an automated test until the feature is more generally released. 
3. Could we file it as a work to do?
4. Cmdlet was extensively tested manually

**Comment by markcowl** [comment]
Issue for test automation here: #3899

**Comment by markcowl** [comment]
Issue for test automation here: #3899

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/NetworkWatcherAPITests.cs (hunk 3 above)
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/NetworkWatcherAPITests.cs]: ssue for test automation here: #3899

File: src/ResourceManager/Network/Commands.Network/help/Test-AzureRmNetworkWatcherConnectivity.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+## OUTPUTS
+
+### Microsoft.Azure.Commands.Network.Models.PSConnectivityInformation
+
+
+## NOTES
+Keywords: azure, azurerm, arm, resource, connectivity, management, manager, network, networking, network watcher
+
+## RELATED LINKS
+[New-AzureRmNetworkWatcher]()
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/help/Test-AzureRmNetworkWatcherConnectivity.md]: Needs links

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1550/

LGTM once this passes

**Pull request merged**